src/core/hle/service/sm/sm.cpp

```
@@ -103,6 +103,8 @@ void SM::GetService(Kernel::HLERequestContext& ctx) {
         rb.Push(client_port.Code());
         LOG_ERROR(Service_SM, "called service=%s -> error 0x%08X", name.c_str(),
                   client_port.Code().raw);
+        if (name.length() == 0)
```

Should we return the error code here that happens on actual HW?

---

(and then remove UNIMPLEMENTED altogether)

---

Might want to change the above `LOG_ERROR` to `LOG_CRITICAL` for extra visibility

---

Discussed over Discord... We'll leave this as-is for now, since we want the UNIMPLEMENTED to halt execution for truly unimplemented services